curator/utils.py

```diff
@@ -748,27 +745,25 @@ def get_client(**kwargs):
 
     try:
         from requests_aws4auth import AWS4Auth
-        kwargs['aws_key'] = False if not 'aws_key' in kwargs \
```

Comment:
Question.  If a user is currently using `aws_key` and `aws_secret_key`, won't this result in a breaking change to them if they upgrade?  I'm concerned about breaking someone's existing functionality in a minor version change.

This would be an easy change for a major revision, but if there's no way to preserve existing functionality, that's going to be a problem.

Reply:
This makes sense. I have added an update to allow for backward compatibility for any existing functionality in a minor version change.

Reply:
I'm not seeing it yet.  Where is it?  It should still allow for these args you've removed.  It has to be there, in case users are running from a non-AWS location (hey, it could happen).

Reply:
Actually, this PR is making credentials handling more standardised because it's delegated to *boto3* `Session`. There are [many ways](http://boto3.readthedocs.io/en/latest/guide/configuration.html) how user can configure credentials - environment variables, profiles, STS credentials,...

But, I definitely agree there should be backward compatibility and maybe deprecate options with next major release.